### PR TITLE
I2s

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: src/i2s0.rs [hunk 1]
```diff
@@ -1,12 +1,166 @@
...
+    pub tor: self::channel::TOR,
+    #[doc = "0x28 - Receive FIFO Configuration Register"]
+    pub rfcr: self::channel::RFCR,
+    #[doc = "0x2c - Transmit FIFO Configuration Register"]
+    pub tfcr: self::channel::TFCR,
+    #[doc = "0x30 - Receive FIFO Flush Register"]
+    pub rff: self::channel::RFF,
+    #[doc = "0x34 - Transmit FIFO Flush Register"]
+    pub tff: self::channel::TFF,
+    #[doc = "0x38 - _RESERVED0"]
```
Comment (burrbull) [inline src/i2s0.rs:71]: Small hack.

File: k210.svd [hunk 2]
```diff
@@ -1319,9 +1319,620 @@
...
-                    <description>Dummy register: this peripheral is not implemented yet</description>
+                    <name>ier</name>
+                    <description>Enable Register</description>
                     <addressOffset>0x00</addressOffset>
+                    <fields>
+                        <field>
+                            <name>ien</name>
+                            <description>I2S Enable</description>
+                            <bitRange>[0:0]</bitRange>
+                            <enumeratedValues>
```
Comment (Disasm) [inline k210.svd]: I doubt that it is useful to have enumerated values for one-bit "enables".

File: k210.svd [hunk 3]
```diff
@@ -1319,9 +1319,620 @@
...
+                    <name>irer</name>
+                    <description>Receiver Block Enable Register</description>
+                    <addressOffset>0x04</addressOffset>
+                    <fields>
+                        <field>
+                            <name>rxen</name>
+                            <description>Receiver block enable</description>
+                            <bitRange>[0:0]</bitRange>
+                            <enumeratedValues derivedFrom="ier.ien.ENABLE">
+                            </enumeratedValues>
```
Comment (Disasm) [inline k210.svd]: Here we have reusing the same enumerated values, but with different meaning: "receiver enabled" vs "i2s enabled". Maybe it is suitable for other use-cases, but not for Rust.

File: k210.svd [hunk 4]
```diff
@@ -1319,9 +1319,620 @@
...
+                    <addressOffset>0x10</addressOffset>
+                    <fields>
+                        <field>
+                            <name>clk_gate</name>
+                            <description>Gating of sclk</description>
+                            <bitRange>[2:0]</bitRange>
+                            <enumeratedValues>
+                                <enumeratedValue>
+                                    <name>no</name>
+                                    <description>Clock gating is disable</description>
```
Comment (Disasm) [inline k210.svd]:
```suggestion
                                    <description>Clock gating is disabled</description>
```

File: k210.svd [hunk 5]
```diff
@@ -1319,9 +1319,620 @@
...
+                    <register>
+                        <name>ter</name>
+                        <description>Transmit Enable Register</description>
+                        <addressOffset>0x0c</addressOffset>
+                        <fields>
+                            <field>
+                                <name>txchenx</name>
+                                <description>Transmit channel enable/disable</description>
+                                <bitRange>[0:0]</bitRange>
+                                <enumeratedValues derivedFrom="rer.rxchenx.ENABLE">
```
Comment (Disasm) [inline k210.svd]: Same issue with enum reuse.

**Review by Disasm — CHANGES_REQUESTED** [review]
Thanks for contributing! There are some debatable moments, however. I can merge the PR as soon as we agree on them.

**Review by Disasm — APPROVED** [review]

**Pull request merged**